src/riak_ql_parser.yrl

```diff
@@ -1210,7 +1213,7 @@ assert_desc_key_field_type(DDL, ?SQL_PARAM{ name = [Name] }) ->
         _ ->
             return_error_flat(
                 "Elements in the local key marked descending (DESC) must be of "
-                "type sint64 or varchar, but was ~p.", [Type])
+                "an integer or binary type, but was ~p.", [Type])
```

Comment:
I prefer general terms so we don't have to keep updating the error message and related tests as we add types (and the feeling was accentuated by the fact that we had forgotten timestamp).

This is a weak preference, however, so I can be talked out of it.